spark-plugins/src/main/java/co/cask/hydrator/plugin/batch/spark/DecisionTreeRegressor.java

```diff
@@ -0,0 +1,225 @@
+import org.apache.twill.filesystem.Location;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import javax.ws.rs.Path;
+
+/**
+ * SparkCompute that uses a trained model to tag input records using Decision Tree regression.
+ */
+@Plugin(type = SparkCompute.PLUGIN_TYPE)
+@Name(DecisionTreeRegressor.PLUGIN_NAME)
+@Description("Uses a trained Decision Tree Regression model and regress records.")
+public class DecisionTreeRegressor extends SparkCompute<StructuredRecord, StructuredRecord> {
+  public static final String PLUGIN_NAME = "DecisionTreeRegressor";
+  private DecisionTreeRegressorConfig config;
+  private Schema outputSchema;
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws IllegalArgumentException {
+    StageConfigurer stageConfigurer = pipelineConfigurer.getStageConfigurer();
+    Schema inputSchema = stageConfigurer.getInputSchema();
+    // if null, the input schema is unknown, or has multiple schemas.
+    if (inputSchema == null) {
+      outputSchema = null;
+      stageConfigurer.setOutputSchema(null);
+      return;
+    }
+    config.validate(inputSchema);
+    // otherwise, we have a constant input schema. Get the input schema and
+    // add a field to it, on which the prediction will be set
+    outputSchema = getOutputSchema(inputSchema);
+    stageConfigurer.setOutputSchema(outputSchema);
+  }
+
+
+  @Override
+  public JavaRDD<StructuredRecord> transform(SparkExecutionPluginContext context,
+                                             JavaRDD<StructuredRecord> input) throws Exception {
+    FileSet fileSet = context.getDataset(config.fileSetName);
+    Location modelLocation = fileSet.getBaseLocation().append(config.path);
+    if (!modelLocation.exists()) {
+      throw new IllegalArgumentException(String.format("Failed to find model to use for Regression. Location does " +
+                                                         "not exist: {}.", modelLocation));
+    }
+
+    final HashMap<String, Map<Object, Long>> categoricalFeaturesMap = new HashMap<>();
+    final String[] fields = config.features.split(",");
+
+    for (final String field : fields) {
+      Map<Object, Long> map = input.map(new Function<StructuredRecord, Object>() {
+        @Override
+        public Object call(StructuredRecord structuredRecord) throws Exception {
+          Schema schema = structuredRecord.getSchema().getField(field).getSchema();
+          if (structuredRecord.get(field) == null && !schema.isNullable()) {
+            throw new IllegalArgumentException(String.format("null value found for non-nullable field %s", field));
+          }
+          Schema.Type type = schema.isNullable() ? schema.getNonNullable().getType() : schema.getType();
+          if (!(type.equals(Schema.Type.INT) || type.equals(Schema.Type.DOUBLE) || type.equals(Schema.Type.FLOAT) ||
```

Comment:
Updated code, to check STRING and BOOLEAN using the type.equals().
BYTES will be directly converted to double when creating the dense vector list. Since there can be too many values for this, and creating index may not be the best approach, therefore, not handling BYTES using type.equals().